pubkey/option.go

```diff
@@ -33,34 +33,34 @@ var (
 	}
 )
 
-// Option represents a functional options pattern interface
+// Option represents a functional option
 type Option func(*pubkeyd) error
 
-// WithAthenzURL represents a AthenzURL functional option
+// WithAthenzURL returns an AthenzURL functional option
 func WithAthenzURL(url string) Option {
-	return func(c *pubkeyd) error {
+	return func(p *pubkeyd) error {
 		if url == "" {
 			return nil
 		}
-		c.athenzURL = regex.ReplaceAllString(url, "")
```

Comment:
@WindzCUHK 

If you delete this logic, there is a possibility that URLs like `https: // http: // example.com` will be generated by processing the following link, you should not delete it.
https://github.com/yahoojapan/athenz-authorizer/blob/master/pubkey/daemon.go#L242

Or I think that it is good to change to logic like the comment of the following link.
https://github.com/yahoojapan/athenz-authorizer/pull/25/commits/b8824f24558262cd646737bcd9b54450d42a232f#diff-debfc42544f659097af6912eacb72a60R37

Reply:
This is also a sub-component.
The logic should be `authorizerd` (the parent) only, or all components.
Which one do you think is better?